src/resolvers/ResolverGenerator.ts

```diff
@@ -0,0 +1,68 @@
+import { readFileSync } from 'fs';
+import { GlobSync } from 'glob';
+import * as handlebars from 'handlebars';
+import { basename, join } from 'path';
+import { ResolverInstance } from '../resolvers/ResolverInstance';
+import { HandlebarsHelpers } from '../schema/Helpers';
+
+/**
+ * Generate Typescript compatible schema
+ */
+export class GraphQLResolverGenerator {
+  private rootTemplate: string = `{{> resolvers}}`
+
+  /**
+   * Register helper functions for Handlebars
+   */
+  constructor() {
+    new HandlebarsHelpers().registerHelpers()
+  }
+
+  /**
+   * Generate Typescript compatible schema
+   * @param resolvers Array of ResolverInstance
+   */
+  public generateResolvers(resolvers: ResolverInstance[]): string {
+    this.registerPartials()
+
+    const template = handlebars.compile(this.rootTemplate)
+    
+    const input: any = {}
+    input.resolvers = this.splitByTypes(resolvers)
+
+    return template(input)
+  }
+
+  /**
+   * Split resolvers into Query and Mutations
+   * @param resolvers Array of ResolverInstance
+   */
+  private splitByTypes(resolvers: ResolverInstance[]) {
+    const computedResolvers = { query: [], mutation: [] };
+    computedResolvers.mutation = resolvers.filter((value: ResolverInstance) => {
+      return value.resolverType === "Mutation"
+    })
+    computedResolvers.query = resolvers.filter((value: ResolverInstance) => {
+      return value.resolverType === "Query"
+    })
+
+    return computedResolvers;
+  }
+
+  /**
+   * Read partials for the template
+   */
+  private registerPartials() {
```

Comment:
DRY - this is helper method that has nothing to do with generators. It is very generic and probably should not be here.

Reply:
Doesn't this read the handlebars templates from the directory?

Reply:
We do not need handlebars for the moment.